src/mod_pushoff_fcm.erl

```diff
@@ -0,0 +1,336 @@
+%%%
+%%% Copyright (C) 2015  Christian Ulrich
+%%% Copyright (C) 2016  Vlad Ki
+%%%
+%%% This program is free software; you can redistribute it and/or
+%%% modify it under the terms of the GNU General Public License as
+%%% published by the Free Software Foundation; either version 2 of the
+%%% License, or (at your option) any later version.
+%%%
+%%% This program is distributed in the hope that it will be useful,
+%%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+%%% General Public License for more details.
+%%%
+%%% You should have received a copy of the GNU General Public License along
+%%% with this program; if not, write to the Free Software Foundation, Inc.,
+%%% 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+%%%
+
+%% This implements the "legacy" binary API
+
+-module(mod_pushoff_fcm).
+
+-author('christian@rechenwerk.net').
+-author('proger@wilab.org.ua').
+-author('dimskii123@gmail.com').
+
+-behaviour(gen_server).
+-compile(export_all).
+-export([init/1,
+         handle_info/2,
+         handle_call/3,
+         handle_cast/2,
+         terminate/2,
+         code_change/3]).
+
+-include("logger.hrl").
+
+-define(SSL_TIMEOUT, 3000).
+-define(MAX_PAYLOAD_SIZE, 2048).
+-define(MESSAGE_EXPIRY_TIME, 86400).
+-define(MESSAGE_PRIORITY, 10).
+-define(MAX_PENDING_NOTIFICATIONS, 1).
+-define(PENDING_INTERVAL, 3000).
+-define(RETRY_INTERVAL, 30000).
+-define(PUSH_URL, "https://fcm.googleapis.com/fcm/send").
+
+-record(state,
+        {certfile :: string(),
+         out_socket :: ssl:socket(),
+         pending_list :: [{pos_integer(), any()}],
+         send_queue :: queue:queue(any()),
+         retry_list :: [any()],
+         pending_timer :: reference(),
+         retry_timer :: reference(),
+         pending_timestamp :: erlang:timestamp(),
+         retry_timestamp :: erlang:timestamp(),
+         message_id :: pos_integer(),
+         gateway :: string(),
```

Comment:
using this value here https://github.com/proger/mod_pushoff/pull/1/commits/8fdbb209b85af9345783f2b265d7eba634a9b6a0#diff-d210ce2737a62db5edf606a09d8e5146R148